tests/core/test_utils.py

```diff
@@ -123,7 +123,6 @@ async def test_error_handling_connection_timeout(caplog):
     assert str(e.value) == str(ONE_WORKING_ONE_TIMEOUT_OUTPUT)
 
     logs = caplog.get_records("call")
-    assert len(logs) == 3
```

Comment:
I think logging from tests running parallel may have been affecting this

Reply:
Surely this should only be a problem if a task in one test isn't torn down correctly. What are the other logs here?

Reply:
Yeah that's right. There are a few tests which aren't being torn down correctly, but I think that has been the case for a while. If you run pytest against main you can see all of this.

```
Task was destroyed but it is pending!
source_traceback: Object created at (most recent call last):
  File "/usr/local/lib/python3.11/threading.py", line 1002, in _bootstrap
    self._bootstrap_inner()
  File "/usr/local/lib/python3.11/threading.py", line 1045, in _bootstrap_inner
    self.run()
  File "/usr/local/lib/python3.11/threading.py", line 982, in run
    self._target(*self._args, **self._kwargs)
  File "/usr/local/lib/python3.11/asyncio/base_events.py", line 608, in run_forever
    self._run_once()
  File "/usr/local/lib/python3.11/asyncio/base_events.py", line 1928, in _run_once
    handle._run()
  File "/usr/local/lib/python3.11/asyncio/events.py", line 84, in _run
    self._context.run(self._callback, *self._args)
  File "/workspaces/ophyd-async/src/ophyd_async/core/_detector.py", line 299, in prepare
    self._arm_status = await self.controller.arm(
  File "/workspaces/ophyd-async/src/ophyd_async/sim/demo/_pattern_detector/_pattern_detector_controller.py", line 33, in arm
    task = asyncio.create_task(
  File "/usr/local/lib/python3.11/asyncio/tasks.py", line 384, in create_task
    task = loop.create_task(coro)
task: <Task pending name='Task-17069' coro=<PatternDetectorController._coroutine_for_image_writing() done, defined at /workspaces/ophyd-async/src/ophyd_async/sim/demo/_pattern_detector/_pattern_detector_controller.py:51> wait_for=<Future pending cb=[Task.task_wakeup()] created at /usr/local/lib/python3.11/asyncio/base_events.py:428> cb=[AsyncStatusBase._run_callbacks()] created at /usr/local/lib/python3.11/asyncio/tasks.py:384>
Task was destroyed but it is pending!
source_traceback: Object created at (most recent call last):
  File "/usr/local/lib/python3.11/threading.py", line 1002, in _bootstrap
    self._bootstrap_inner()
  File "/usr/local/lib/python3.11/threading.py", line 1045, in _bootstrap_inner
    self.run()
  File "/usr/local/lib/python3.11/threading.py", line 982, in run
    self._target(*self._args, **self._kwargs)
  File "/usr/local/lib/python3.11/asyncio/base_events.py", line 608, in run_forever
    self._run_once()
  File "/usr/local/lib/python3.11/asyncio/base_events.py", line 1928, in _run_once
    handle._run()
  File "/usr/local/lib/python3.11/asyncio/events.py", line 84, in _run
    self._context.run(self._callback, *self._args)
  File "/workspaces/ophyd-async/src/ophyd_async/core/_detector.py", line 299, in prepare
    self._arm_status = await self.controller.arm(
  File "/workspaces/ophyd-async/src/ophyd_async/sim/demo/_pattern_detector/_pattern_detector_controller.py", line 33, in arm
    task = asyncio.create_task(
  File "/usr/local/lib/python3.11/asyncio/tasks.py", line 384, in create_task
    task = loop.create_task(coro)
task: <Task pending name='Task-17015' coro=<PatternDetectorController._coroutine_for_image_writing() running at /workspaces/ophyd-async/src/ophyd_async/sim/demo/_pattern_detector/_pattern_detector_controller.py:56> wait_for=<Future pending cb=[Task.task_wakeup()] created at /usr/local/lib/python3.11/asyncio/base_events.py:428> cb=[AsyncStatusBase._run_callbacks()] created at /usr/local/lib/python3.11/asyncio/tasks.py:384>
Task was destroyed but it is pending!
source_traceback: Object created at (most recent call last):
  File "/usr/local/lib/python3.11/threading.py", line 1002, in _bootstrap
    self._bootstrap_inner()
  File "/usr/local/lib/python3.11/threading.py", line 1045, in _bootstrap_inner
    self.run()
  File "/usr/local/lib/python3.11/threading.py", line 982, in run
    self._target(*self._args, **self._kwargs)
  File "/usr/local/lib/python3.11/asyncio/base_events.py", line 608, in run_forever
    self._run_once()
  File "/usr/local/lib/python3.11/asyncio/base_events.py", line 1928, in _run_once
    handle._run()
  File "/usr/local/lib/python3.11/asyncio/events.py", line 84, in _run
    self._context.run(self._callback, *self._args)
  File "/workspaces/ophyd-async/src/ophyd_async/core/_detector.py", line 299, in prepare
    self._arm_status = await self.controller.arm(
  File "/workspaces/ophyd-async/src/ophyd_async/sim/demo/_pattern_detector/_pattern_detector_controller.py", line 33, in arm
    task = asyncio.create_task(
  File "/usr/local/lib/python3.11/asyncio/tasks.py", line 384, in create_task
    task = loop.create_task(coro)
task: <Task pending name='Task-17045' coro=<PatternDetectorController._coroutine_for_image_writing() running at /workspaces/ophyd-async/src/ophyd_async/sim/demo/_pattern_detector/_pattern_detector_controller.py:56> wait_for=<Future pending c
```

I don't know why this PR caused it to error, but I'm guessing any old change might have caused this at some point? I can write another issue for it

Reply:
I suspect there is some `AsyncStatus` not being directly awaited, and so being garbage collected later on. `test_status.py` is a good candidate based on it's proximity to `test_utils.py` in test execution order.